app/controllers/criteria_controller.rb

```diff
@@ -115,9 +115,15 @@ def update
   def update_positions
     @assignment = Assignment.find(params[:assignment_id])
 
-    ApplicationRecord.transaction do
+    Criterion.transaction(requires_new: true) do
       params[:criterion].each_with_index do |id, index|
-        Criterion.update(id, position: index + 1) unless id.blank?
+        begin
+          @assignment.criteria.find(id)
```

Comment:
`find` returns a criterion. Save it in a variable so you can call `update` on it on the following line.